### PR TITLE
Fix "Failed to fetch local disk feature on Windows"

### PR DESCRIPTION
**Pull request by joshspicer**
fixes: https://github.com/devcontainers/cli/issues/321

Also adds in a test to prevent regression via a new Windows-based test workflow. 

NOTE: Refactoring/updating the test suite will need to be done incrementally to get 100% of the test suite running (at that point we can remove the new workflow and matrix running the tests against `ubuntu-latest` and `windows-latest` in a single workflow)

`shelljs` is added a dev dependency to abstract shell commands when running on linux/windows (see this closed PR for an example of how it will be used: https://github.com/devcontainers/cli/pull/177)

### CODE REVIEW

File: src/spec-configuration/containerFeaturesConfiguration.ts [hunk 1]
```diff
@@ -547,9 +556,16 @@ function featuresToArray(config: DevContainerConfig, additionalFeatures: Record<
 
 // Process features contained in devcontainer.json
 // Creates one feature set per feature to aid in support of the previous structure.
-async function processUserFeatures(output: Log, config: DevContainerConfig, workspaceRoot: string, env: NodeJS.ProcessEnv, userFeatures: DevContainerFeature[], featuresConfig: FeaturesConfig, skipFeatureAutoMapping: boolean): Promise<FeaturesConfig> {
+async function processUserFeatures(params: ContainerFeatureInternalParams, config: DevContainerConfig, workspaceRoot: string, userFeatures: DevContainerFeature[], featuresConfig: FeaturesConfig): Promise<FeaturesConfig> {
+	const { platform, output, skipFeatureAutoMapping, env, } = params;
+
+	let configPath = config.configFilePath.path;
```
Comment (chrmarti) [inline src/spec-configuration/containerFeaturesConfiguration.ts:562]: `config.configFilePath.fsPath` should give you the local path.

**Review by chrmarti — CHANGES_REQUESTED** [review]
Left a comment for simplifying the code. Thanks.

File: src/spec-configuration/containerFeaturesConfiguration.ts (hunk 1 above)
Comment (joshspicer) [reply src/spec-configuration/containerFeaturesConfiguration.ts:562]: In codespaces, it gives a `vscode-remote://` URI - is that what we want?

File: package.json [hunk 2]
```diff
@@ -84,6 +85,7 @@
 		"npm-run-all": "^4.1.5",
 		"p-all": "^4.0.0",
 		"rimraf": "^3.0.2",
+		"shelljs": "^0.8.5",
```
Comment (chrmarti) [inline package.json:88]: Is that needed? Maybe a leftover?

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (joshspicer) [reply package.json:88]: I am eventually planning on using this for getting the tests to run cross platform (see PR description), but yea not needed yet so I will remove 👍

**Review by chrmarti — APPROVED** [review]
👍

**Pull request merged**